<?xml version="1.0"?>
<launch>

    <arg name="target_frame" default="base_link" />
    <arg name="marker_length" default="0.04" /> <!-- in meters -->
    <arg name="retry_attempts" default="3" />
    <arg name="debug" default="true" />
    <arg name="output_object_list" default="/mcr_perception/object_list_merger/output_object_list" />

    <node pkg="mir_perceive_aruco_cube" type="aruco_cube_perceiver" name="aruco_cube_perceiver"
          output="screen" ns="mir_perception"> 
        <remap from="~input_image" to="/arm_cam3d/rgb/image_raw"/> 
        <remap from="~input_pointcloud" to="/arm_cam3d/depth_registered/points"/> 
        <remap from="~obj_list" to="$(arg output_object_list)"/> 

        <param name="target_frame" type="str" value="$(arg target_frame)"/> 
        <param name="num_of_retries" type="int" value="$(arg retry_attempts)"/> 
        <param name="debug" type="bool" value="$(arg debug)"/> 
    </node> 
</launch>
